### DISCUSSION TITLE
identifier not found

### DISCUSSION THREAD

**Discussion by HernanDarioCostantini** [virtual-maker/VirtualBoard#4] [Q&A] [answered]
Hi. I´m trying to run this simple code:
/////////////////////////////////////////////////////////////////////////////////////
void setup()
{
	Serial.begin(115200);
	Serial.println("Hello, World!");
}
// Pay attention that int is int32_t with MSVC
// This differs from e.g. Arduino Uno where int is int16_t
int c = 0; 

void loop()
{
	Serial.println(c);
	c++;
	delay(500);
	something();
}
void something(){
	c = 10;
}
/////////////////////////////////////////////////////////////////////////////////////////////////

but I m getting:

C3861  'something': identifier not found

It´s so simple that I have no idea if it´s me or it is just a bug.........
any ideas?

**Comment by virtual-maker** [comment, marked as answer]
@HernanArg2022Beginner
Other than the Arduino IDE, VirtualBoard uses proper C++. Therefore, a function must be declared or defined before it is used the first time. First time refers to processing the C++ file top to bottom.

- Either move the function before setup() or loop() or
- Add a forward declaration.

This is same issue like with PlatformIO, see: 
[How to create a custom method in Arduino](https://community.platformio.org/t/how-to-create-a-custom-method-in-arduino/10554)
[Convert Arduino file to C++ manually](https://docs.platformio.org/en/latest/faq/ino-to-cpp.html)
